### ISSUE TITLE
teams-recording-bot won't start running | _botService.Initialize ();

### ISSUE THREAD

**Issue by dejmsi** [microsoftgraph/microsoft-graph-comms-samples#397]
the query is directed to the project - https://github.com/microsoftgraph/microsoft-graph-comms-samples/tree/master/Samples/V1.0Samples/AksSamples/teams-recording-bot

I try to run this example, I have everything I need for it and I followed the instructions, though. some points come to me tossed. Does anyone have experience with this and did you manage to run it into a running shoe? 

My current result is that after setting all points 1-7 and running the program in Visual Studio, it crashes due to an error on /RecordingBot.Services/ServiceSetup/AppHost.cs on the line `_botService.Initialize ();` , where I could not solve this problem.

I should have all the parameters from Azure, NgRok and Certificate set correctly.

**Comment by 1fabi0** [comment]
Can you please share the exact issue it is throwing?

**Comment by dejmsi** [comment]
specifically, it writes this in the figure "The object reference is not set to an object instance".

It is possible that I have something wrong stated during the setup, such as certificates or ngrok. if you would like to advise me, you can go to dejmsi (a) gmail.com, I will be grateful.

![image](https://user-images.githubusercontent.com/11996991/105863144-d8432180-5ff0-11eb-979b-68b0de432883.png)

**Comment by 1fabi0** [comment]
It seems like you didn't call boot server so the dependency can be resolved

**Comment by umkhan65** [comment]
@1fabi0 did you find out what was the issue?

**Comment by 1fabi0** [comment]
@umkhan65 as I said this happens because boot isn't called so the `IServiceCollection` isn't added with all needed classes and the `IServiceProvider` Can't resolve the dependency and returns `null` on the `GetService` Method If you would use `GetRequiredService` you will end up with a Exception on service Resolving

**Comment by JasonTheDeveloper** [comment]
It's almost as if `_botService` isn't resolved properly. Can you pop a breakpoint on line [101](https://github.com/microsoftgraph/microsoft-graph-comms-samples/blob/master/Samples/V1.0Samples/AksSamples/teams-recording-bot/src/RecordingBot.Services/ServiceSetup/AppHost.cs#L101) and see if it throws any errors?

**Comment by dejmsi** [comment]
yes, thank you for the news. The problem was in the certificates, which was probably not well created, currently the application was running fine for me. In my opinion, this is not completely described in the documentation and can be confusing at times.

**Comment by beth-panx** [comment]
Experiencing the same error with ngrok returning a /502 bad gateway... 

Followed the instruction in this repo step by step and didn't modify anything.

**Comment by 1fabi0** [comment]
I think if you experience problems with ngrok best you can do is searching for solutions in the ngrok repository https://github.com/inconshreveable/ngrok
